pkg/arvo/app/hark-graph-hook.hoon

```diff
@@ -0,0 +1,258 @@
+      (need (peek-metadata:met %graph group resource.q.update))
+    =*  rid  resource.q.update
+    =+  %+  scry:ha
+           ,mark=(unit mark) 
+        /gx/graph-store/graph-mark/(scot %p entity.rid)/[name.rid]/noun
+    =+  %+  scry:ha
+          ,=tube:clay
+        /cc/[q.byk.bowl]/[(fall mark %graph-validator-link)]/notification-kind
+    =/  nodes=(list [p=index:graph-store q=node:graph-store])
+      ~(tap by nodes.q.update)
+    =|  cards=(list card)
+    |^
+    ?~  nodes 
+      [cards state]
+    =*  index  p.i.nodes
+    =*  node   q.i.nodes
+    =^  node-cards  state
+      (check-node node tube)
+    %_    $
+      nodes  t.nodes
+      cards  (weld node-cards cards)
+    ==
+    ::
+    ++  check-node-children
+      |=  [=node:graph-store =tube:clay]
+      ^-  (quip card _state)
+      ?:  ?=(%empty -.children.node)
+        [~ state]
+      =/  children=(list [=atom =node:graph-store])
+        (tap:orm:graph-store p.children.node)
+      =|  cards=(list card)
+      |-  ^-  (quip card _state)
+      ?~  children
+        [cards state]
+      =^  new-cards  state
+        (check-node node.i.children tube)
+      %_    $
+        cards  (weld cards new-cards) 
+        children  t.children
+      ==
+    ::
+    ++  check-node
+      |=  [=node:graph-store =tube:clay]
+      ^-  (quip card _state)
+      =^  child-cards  state
+        (check-node-children node tube)
+      ?:  =(our.bowl author.post.node)
+        =^  self-cards  state
+          (self-post node)
+        :_  state
+        (weld child-cards self-cards)
+      =+  !<(notif-kind=(unit @t) (tube !>([0 post.node])))
+      ?~  notif-kind
+        [child-cards state]
+      =/  desc=@t
+        ?:  (is-mention contents.post.node)
+          %mention 
+        u.notif-kind
+      ?.  ?|  =(desc %mention)
+              (~(has in watching) rid)
+          ==
+        [child-cards state]
+      =/  notif-index=index:store
+        [%graph group rid module.metadata desc]
+      =/  =contents:store
+        [%graph (limo post.node ~)]
+      :_  state
+      %+  snoc  child-cards
+      (add-unread notif-index [time-sent.post.node %.n contents])
+    ::
+    ++  is-mention
+      |=  contents=(list content:post)
+      ^-  ?
+      ?.  mentions  %.n
+      ?~  contents  %.n
+      ?.  ?=(%mention -.i.contents)
+        $(contents t.contents)
+      ?:  =(our.bowl ship.i.contents)
+        %.y
+      $(contents t.contents)
+    ::
+    ++  self-post
+      |=  =node:graph-store
+      ^-  (quip card _state)
+      ?.  ?=(%.y watch-on-self)
+        [~ state]
+      `state(watching (~(put in watching) rid))
+    ::
+    ++  add-unread
+      |=  [=index:store =notification:store]
+      ^-  card 
+      =-  [%pass / %agent [our.bowl %hark-store] %poke -]
+      hark-action+!>([%add index notification])
+    ::
+    --
+  --
+::
+++  on-peek  on-peek:def
+::
+++  on-leave  on-leave:def
+++  on-arvo  on-arvo:def
+++  on-fail   on-fail:def
+--
+|_  =bowl:gall
```

Comment:
I think this should be declared above the agent core with a `=>` given that it's the shorter of the two cores.